### PR TITLE
scripts/sign_encrypt.py: Support verification of a signed TA

### PR DESCRIPTION
**Pull request by hoiho-amzn**
Adds a new option 'verify' to sign_encrypt.py to verify whether a
Trusted Application is signed correctly.

Required arguments: --uuid, --in, --key

Signed-off-by: Donald Chan <hoiho@amazon.com>

### CODE REVIEW

File: scripts/sign_encrypt.py [hunk 1]
```diff
@@ -327,13 +329,81 @@ def stitch_ta():
...
+
+        start, end = end, SHDR_SIZE + digest_len + sig_len
+        signature = img[start:end]
+
+        # Extract UUID and TA version
+        start, end = end, end + 16 + 4
+        [uuid, ta_version] = struct.unpack('<16sI', img[start:end])
+
+        if magic != SHDR_MAGIC:
+            raise Exception("Un-expected magic: 0x{:08x}".format(magic))
```
Comment (jforissier) [inline scripts/sign_encrypt.py:353]: Unexpected

File: scripts/sign_encrypt.py [hunk 2]
```diff
@@ -327,13 +329,81 @@ def stitch_ta():
...
+
+        if img_type != SHDR_BOOTSTRAP_TA:
+            raise Exception("Un-supported image type: {}".format(img_type))
```
Comment (jforissier) [inline scripts/sign_encrypt.py:356]: Unsupported

File: scripts/sign_encrypt.py [hunk 3]
```diff
@@ -327,13 +329,81 @@ def stitch_ta():
...
+
+        if algo_value not in algo.values():
+            raise Exception('Un-recognized algorithm: 0x{:08x}'
```
Comment (jforissier) [inline scripts/sign_encrypt.py:359]: Unrecognized

**Review by jforissier — COMMENTED** [review]
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by hoiho-amzn** [comment]
All comments have been addressed.

**Comment by jforissier** [comment]
> All comments have been addressed.

You applied my Ack but you didn't change the content of the patch (forgot `git add`?). That's bad.

**Comment by hoiho-amzn** [comment]
Oops - my bad. I forgot to `git add` - it's back there now.

**Pull request merged**